lib/kafo/data_type_parser.rb

```diff
@@ -8,9 +8,26 @@ class DataTypeParser
 
     def initialize(manifest)
       @logger = KafoConfigure.logger
-      @types = {}
+
+      lines = []
+      last_line = ''
       manifest.each_line do |line|
-        if (type = TYPE_DEFINITION.match(line.force_encoding("UTF-8")))
+        line = line.force_encoding("UTF-8").strip
+        next if line.start_with?('#')
+
+        line = line.split('#').first.strip
```

Comment:
This will break if you use a Pattern with `#` in it.

Reply:
So will the current regex, no?

https://github.com/theforeman/kafo/blob/4b44b57ca54f3fb6cc1e46ed511093a6dd61f519/lib/kafo/data_type_parser.rb#L5

Reply:
However, https://www.puppet.com/docs/puppet/7/lang_comments doesn't say anything about leading spaces

Reply:
added a test for the *one* Pattern I could find using a `#` inside, `Stdlib::Email` :)